src/base/bittorrent/session.cpp

```diff
@@ -1447,11 +1447,13 @@ void Session::configureNetworkInterfaces(lt::settings_pack &settingsPack)
         if (!addr.isNull())
         {
             const QString ip = ((addr.protocol() == QAbstractSocket::IPv6Protocol)
-                          ? ('[' + Utils::Net::canonicalIPv6Addr(addr).toString() + ']')
-                          : addr.toString());
-            endpoints << (ip + portString);
+                          ? (Utils::Net::canonicalIPv6Addr(addr).toString())
+                          : addr.toString());;
+            endpoints << ((addr.protocol() == QAbstractSocket::IPv6Protocol)
+                          ? QString('[' + ip + ']') + portString
+                          : ip + portString);
 
-            if ((ip != "0.0.0.0") && (ip != "[::]"))
+            if ((ip != "0.0.0.0") && (ip != "::"))
```

Comment:
Pardon me, I couldn't really find the difference of your changes except this very line.
It seems to me that all changes above this line should be reverted and this line would become:
```c++
if (ip != "0.0.0.0")
```
Would this form solve your problem?

Also please elaborate what is the problem, I looked into the linked issue but didn't found any concrete examples.

Reply:
OK, thanks for the detailed explanation, it is clear now.

I would suggest you take the following code: (please try it first)
```c++
        if (!addr.isNull())
        {
            const bool isIPv6 {addr.protocol() == QAbstractSocket::IPv6Protocol};
            const QString ip {isIPv6
                ? Utils::Net::canonicalIPv6Addr(addr).toString()
                : addr.toString()};

            endpoints << ((isIPv6 ? ('[' + ip + ']') : ip) + portString);
            if ((ip != QLatin1String("0.0.0.0")) && (ip != QLatin1String("::")))
                outgoingInterfaces << ip;
        }
```

Also you are required to squash the all commits into one before we can approve and merge.

Reply:
Done and done. Works as expected.